web-registry/src/features/marketplace/BridgeFlow/hooks/useCreditBridgeSubmit.tsx

```diff
@@ -0,0 +1,127 @@
+import { useCallback } from 'react';
+import { MsgBridge } from '@regen-network/api/lib/generated/regen/ecocredit/v1/tx';
+import { useTrack } from 'use-analytics';
+
+import type { BridgeFormValues } from 'web-components/lib/components/form/BridgeForm';
+import type { Item } from 'web-components/lib/components/modal/TxModal';
+
+import type { BatchInfoWithBalance } from 'types/ledger/ecocredit';
+import type { UseStateSetter } from 'types/react/use-state';
+
+import type { SignAndBroadcastType } from 'hooks/useMsgClient';
+
+import {
+  BRIDGE_BUTTON_TEXT,
+  BRIDGE_HEADER,
+  BRIDGE_TITLE,
+  BRIDGE_TX_DESCRIPTION,
+} from '../BridgeFlow.constants';
+
+type Props = {
+  accountAddress?: string;
+  creditBridgeBatch?: BatchInfoWithBalance;
+  signAndBroadcast: SignAndBroadcastType;
+  setCreditBridgeOpen: UseStateSetter<BatchInfoWithBalance | undefined>;
+  setCardItems: UseStateSetter<Item[] | undefined>;
+  setTxModalHeader: UseStateSetter<string>;
+  setTxModalTitle: UseStateSetter<string>;
+  setTxButtonTitle: UseStateSetter<string>;
+  setTxModalDescription: UseStateSetter<string>;
+};
+
+type ReturnType = (values: BridgeFormValues) => Promise<void>;
+
+const useCreditBridgeSubmit = ({
+  accountAddress,
+  creditBridgeBatch,
+  signAndBroadcast,
+  setCreditBridgeOpen,
+  setCardItems,
+  setTxModalHeader,
+  setTxModalTitle,
+  setTxButtonTitle,
+  setTxModalDescription,
+}: Props): ReturnType => {
+  const track = useTrack();
```

Comment:
I went ahead and added the tracking events, but we can add them separately if this is out of scope. cc @wgwz